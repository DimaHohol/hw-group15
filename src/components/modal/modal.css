.overlay {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  /* z-index: 98; */
  height: 100%;
  width: 100%;
  background-color: rgba(0, 0, 0, 0.3);
  z-index: 99;
}

.modal {
  position: relative;
  z-index: 98;
  width: 100%;
  max-width: 26rem;
  max-height: 100%;
}
.modal-close {
  position: absolute;
  top: -1.5rem;
  right: 0;
  padding: 5px;
  border: 0;
  -webkit-appearance: none;
  background: none;
  color: white;
  cursor: pointer;
}

.modal-body {
  border-radius: 4px;
  background-color: #3c5d50;
  border-radius: 10px;
}

.modal-body-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: gray;
  height: 50px;
}
.modal-body-content {
  display: flex;
  padding: 1.25rem 1.5rem;
}
.modal-body-footer {
  display: flex;
  justify-content: center;
  align-items: center;
}
