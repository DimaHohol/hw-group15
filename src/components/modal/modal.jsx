// react-hw-group15/hw-group15/src/components/modal/modal.jsx

import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { setModalState } from "../../redux/actions";
import "./modal.css";

const Modal = ({
  header,
  closeButton,
  text,
  actions,
  onClose,
  isModalOpen,
}) => {
  const handleClose = () => {
    onClose();

    setModalState(false);
  };

  return (
    <div className={`modal ${isModalOpen ? "open" : ""}`}>
      <div className="modal-content">
        {closeButton && (
          <button className="close-button" onClick={handleClose}>
            X
          </button>
        )}
        <h2 className="modal-header">{header}</h2>
        <div className="modal-text">{text}</div>
        <div className="modal-actions">{actions}</div>
      </div>
    </div>
  );
};

Modal.propTypes = {
  header: PropTypes.string.isRequired,
  closeButton: PropTypes.bool,
  text: PropTypes.string.isRequired,
  actions: PropTypes.node.isRequired,
  onClose: PropTypes.func.isRequired,
  isModalOpen: PropTypes.bool.isRequired,
};

Modal.defaultProps = {
  closeButton: false,
};

const mapStateToProps = (state) => {
  return {
    isModalOpen: state.isModalOpen,
  };
};

export default connect(mapStateToProps, { setModalState })(Modal);
