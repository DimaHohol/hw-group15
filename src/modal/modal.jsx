import React from "react";
import "./modal.css";

class Modal extends React.Component {
  render() {
    const { header, text, actions, onClose, closeButtonX } = this.props;
    return (
      <div>
        <div className="overlay">
          <div className="modal">
            <div className="modal-body">
              <div className="modal-body-header">
                <h3>{header}</h3>
                {closeButtonX}
              </div>
              <div className="modal-body-content">
                <p>{text}</p>
              </div>

              <div className="modal-body-footer">{actions}</div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Modal;
