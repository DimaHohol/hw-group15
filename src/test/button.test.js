import React from "react";
import { render } from "@testing-library/react";
import Button from "../components/button/button";
import renderer from "react-test-renderer";
import modal from "../components/modal/modal";

test("renders component correctly", () => {
  render(<Button />);
});
