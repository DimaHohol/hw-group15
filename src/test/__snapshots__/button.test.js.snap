import React from "react";
import { render } from "@testing-library/react";
import Button from "../components/button/button";
import renderer from "react-test-renderer";
import modal from "../components/modal/modal";


it("renders correctly", () => {
  const tree = renderer
    .create(<Button text="Click me" click={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it("renders correctly", () => {
  const tree = renderer
    .create(
      <Modal
        header="Modal Header"
        closeButton={true}
        text="Modal text"
        actions={<button>Close</button>}
        onClose={() => {}}
        isModalOpen={true}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});